File: C++/617_Merge_Two_Binary_Trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
        if(t1 == NULL) return t2;
        stack<vector<TreeNode*>> stack;
        stack.push({t1, t2});
        while(!stack.empty()){
            auto t = stack.top();
            stack.pop();
            if(t[0] == NULL || t[1] == NULL) continue;
            t[0]->val += t[1]->val;
            if(t[0]->left == NULL) t[0]->left = t[1]->left;
            else stack.push({t[0]->left, t[1]->left});
            if(t[0]->right == NULL) t[0]->right = t[1]->right;
            else stack.push({t[0]->right, t[1]->right});
        }
        return t1;
    }
};

File: Python/617_Merge_Two_Binary_Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def mergeTrees(self, t1: TreeNode, t2: TreeNode) -> TreeNode:
        if t1 != None and t2 != None:
            t = TreeNode(t1.val + t2.val)
            t.left = self.mergeTrees(t1.left, t2.left)
            t.right = self.mergeTrees(t1.right, t2.right)
            return t
        elif t1 != None:
            t = TreeNode(t1.val)
            t.left = self.mergeTrees(t1.left, t2)
            t.right = self.mergeTrees(t1.right, t2)
            return t
        elif t2 != None:
            t = TreeNode(t2.val)
            t.left = self.mergeTrees(t1, t2.left)
            t.right = self.mergeTrees(t1, t2.right)
            return t
